akka-typed/src/main/scala/akka/typed/javadsl/ActorContext.scala

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.javadsl
+
+import java.util.function.{ Function ⇒ JFunction }
+import akka.annotation.DoNotInherit
+import akka.annotation.ApiMayChange
+import akka.typed.ActorRef
+import akka.typed.ActorSystem
+import java.util.Optional
+import akka.typed.Behavior
+import akka.typed.DeploymentConfig
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.ExecutionContextExecutor
+
+/**
+ * An Actor is given by the combination of a [[Behavior]] and a context in
+ * which this behavior is executed. As per the Actor Model an Actor can perform
+ * the following actions when processing a message:
+ *
+ *  - send a finite number of messages to other Actors it knows
+ *  - create a finite number of Actors
+ *  - designate the behavior for the next message
+ *
+ * In Akka the first capability is accessed by using the `tell` method
+ * on an [[ActorRef]], the second is provided by [[ActorContext#spawn]]
+ * and the third is implicit in the signature of [[Behavior]] in that the next
+ * behavior is always returned from the message processing logic.
+ *
+ * An `ActorContext` in addition provides access to the Actor’s own identity (“`getSelf`”),
+ * the [[ActorSystem]] it is part of, methods for querying the list of child Actors it
+ * created, access to [[Terminated DeathWatch]] and timed message scheduling.
+ */
+@DoNotInherit
+@ApiMayChange
+trait ActorContext[T] {
+  // this must be a pure interface, i.e. only abstract methods
+
+  /**
+   * The identity of this Actor, bound to the lifecycle of this Actor instance.
+   * An Actor with the same name that lives before or after this instance will
+   * have a different [[ActorRef]].
+   */
+  def getSelf: ActorRef[T]
+
+  /**
+   * Return the mailbox capacity that was configured by the parent for this actor.
+   */
+  def getMailboxCapacity: Int
+
+  /**
+   * The [[ActorSystem]] to which this Actor belongs.
+   */
+  def getSystem: ActorSystem[Void]
```

Comment:
Hm... Void since we have Nothing in Scala right? Yeah that's likely as close as we can get hm